ami/ml/orchestration/nats_queue.py

```
@@ -330,9 +338,134 @@ async def delete_stream(self, job_id: int) -> bool:
+
+                # Flush to ensure all ACKs are written to the socket before unsubscribing.
+                # msg.ack() only queues a publish in the client buffer; without flush() the
+                # ACKs can be silently dropped when the subscription is torn down.
+                await self.nc.flush()
+
+            except (asyncio.TimeoutError, nats.errors.TimeoutError):
+                logger.info(f"No advisory messages found for job {job_id}")
+            finally:
+                await psub.unsubscribe()
+
+        except Exception as e:
```

Ruff (0.15.5)

[warning] Do not catch blind exception: `Exception`

(BLE001)